### DISCUSSION TITLE
<input type=date> x-model.lazy ignores lazy modifier, updates immediately

### DISCUSSION THREAD

**Discussion by Fak3** [alpinejs/alpine#4378] [7. Bugs (archived)]
When x-model.lazy is used on input with type=date, lazy modifier does not work. Data updates immediately on keypress.
```html
<div x-data="{date: '2024-09-03'}">
  <input type=date x-model.lazy='date'/>
  <div x-text="date"></div><span>BUG: text above ignores lazy <b>date</b> input, updates on keypress</span>
</div>
```
playground: https://jsfiddle.net/vuLsgm20/10/

Expected behavior: lazy modifier must prevent immediate data update on keypress, and wait for focusout event.

alpinejs version: 3.14.1

**Comment by SimoTod** [comment]
x-model updates on the `input` event. the lazy modifier changes the behaviour so it updates on the `change` event.
Unfortunately, the date input fields sends a change event every time you have a complete date at the same time of the input one.
There's no much to do other then mentioning it in the documentation.

**Comment by ekwoka** [comment]
using focusout isn't a catchall solution, since Select elements would still be focused after the user selects an option and the panel closes.

this would need more massaging to see which if any would need to change to blur vs change.
